spec/unit/policy_builder/policyfile_spec.rb

```diff
@@ -379,67 +398,97 @@ def initialize_pb
 
         let(:cookbook_synchronizer) { double("Chef::CookbookSynchronizer") }
 
-        context "and a cookbook is missing" do
+        shared_examples_for "fetching cookbooks" do
+          context "and a cookbook is missing" do
 
-          let(:error404) { Net::HTTPServerException.new("404 message", :body) }
+            let(:error404) { Net::HTTPServerException.new("404 message", :body) }
 
-          before do
-            expect(Chef::Node).to receive(:find_or_create).with(node_name).and_return(node)
+            before do
+              expect(Chef::Node).to receive(:find_or_create).with(node_name).and_return(node)
 
-            # Remove references to example2 cookbook because we're iterating
-            # over a Hash data structure and on ruby 1.8.7 iteration order will
-            # not be stable.
-            parsed_policyfile_json["cookbook_locks"].delete("example2")
-            parsed_policyfile_json["run_list"].delete("recipe[example2::server]")
+              # Remove references to example2 cookbook because we're iterating
+              # over a Hash data structure and on ruby 1.8.7 iteration order will
+              # not be stable.
+              parsed_policyfile_json["cookbook_locks"].delete("example2")
+              parsed_policyfile_json["run_list"].delete("recipe[example2::server]")
```

Comment:
Can you try getting rid of these 5 lines because we're not using ruby 1.8.7 anymore?  We should be able to iterate over the hash in order now.